File: java/3.3/src/pubnub/api/Callback.java
package pubnub.api;

public interface Callback {

	public abstract boolean subscribeCallback(String channel, Object message);

	public abstract boolean presenceCallback(String channel, Object message);
	
	public abstract void errorCallback(String channel, Object message);

	public abstract void connectCallback(String channel);

	public abstract void reconnectCallback(String channel);

	public abstract void disconnectCallback(String channel);
}

File: csharp-silverlight/3.3/PubnubSilverlight/PubnubSilver/src/Callback.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;

namespace silverlight
{
    public interface Callback
    {
        bool responseCallback(string channel, object message);

        void errorCallback(String channel, Object message);

        void connectCallback(String channel);

        void reconnectCallback(String channel);

        void disconnectCallback(String channel);
    }    
}
